### PR TITLE
feat: Add `emath::Number` compatibility and `Default` impl, add `new_masked` and `into_inner` functions

### PR DESCRIPTION
**Pull request by C0D3-M4513R**
### Summary of the PR

*Please summarize here why the changes in this PR are needed.*

To Construct a `egui::DragValue` the type needs to implement `emath::Number`.
This pr adds a feature to add a `emath::Number` implementation, a `Default` implementation and 2 other convenience functions.

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR are signed (with `git commit -s`), and the commit
  message has max 60 characters for the summary and max 75 characters for each
  description line.
- [x] All added/changed functionality has a corresponding unit/integration
  test.
- [x] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [x] Any newly added `unsafe` code is properly documented.

Closes: #29

### CODE REVIEW

**Comment by C0D3-M4513R** [comment]
> All added/changed functionality has a corresponding unit/integration
test.

I'm sorry what unit/integration tests?

**Comment by JonathanWoollett-Light** [comment]
Apologies for delay on reviewing, I've been quite busy lately, I will try to get to this in the next week.

**Comment by C0D3-M4513R** [comment]
uhm sorry, some commits leaked from a rewrite I attempted.
Also I only implemented the changes for unsigned types I realised.
The proc_macro code is quite unreadable to me, so I apologise.

File: ux2-macros/src/lib.rs [hunk 1]
```diff
@@ -823,6 +823,22 @@ pub fn generate_types(item: proc_macro::TokenStream) -> proc_macro::TokenStream
             #[repr(transparent)]
             #[derive(Debug, Eq, PartialEq, Ord, PartialOrd, Clone, Copy, Hash)]
             pub struct #unsigned_ident(#unsigned_inner_ident);
+            #[cfg(feature = "emath")]
```
Comment (JonathanWoollett-Light) [inline ux2-macros/src/lib.rs]: Please add a newline here.

File: ux2-macros/src/lib.rs [hunk 2]
```diff
@@ -823,6 +823,22 @@ pub fn generate_types(item: proc_macro::TokenStream) -> proc_macro::TokenStream
...
+                #[inline(always)]
+                fn to_f64(self) -> f64 {
+                    self.0 as f64
+                }
+
+                #[inline(always)]
+                fn from_f64(num: f64) -> Self {
+                    Self::new(num as #unsigned_inner_ident)
+                }
+            }
```
Comment (JonathanWoollett-Light) [inline ux2-macros/src/lib.rs:841]: Please add a newline here.

File: ux2/Cargo.toml [hunk 3]
```diff
@@ -17,10 +17,12 @@ default = ["8"]
 32 = ["16"]
 16 = ["8"]
 8 = []
+emath = ["dep:emath"]
 
 [dependencies]
 serde = { version = "1.0.160", features = ["derive"], optional = true }
 ux2-macros =  { path = "../ux2-macros", version = "0.9.0" }
+emath = {version = "0.25", optional = true}
```
Comment (JonathanWoollett-Light) [inline ux2/Cargo.toml]: Nit:
```suggestion
emath = { version = "0.25", optional = true }
```

**Review by JonathanWoollett-Light — CHANGES_REQUESTED** [review]
For the `emath` change please squash all changes into 1 commit, entitled `` feat: Add `emath::number` ``, you do not need to change the changelog or readme, this will be done automatically by CI.

The other changes like `Add no_std support` should be in a separate PR.

**Comment by JonathanWoollett-Light** [comment]
> Each? That'll be a ton of extra work, for something that'll be merged regardless, and create unnecessary conflicts in the changelog?

1 PR per area of change is best practices and helps speed up reviews. You don't need to edit the changelog.

**Comment by C0D3-M4513R** [comment]
Okay. I made a pr for each thing.
Please note, that I cannot guarantee the correctness of some of the pr's, as some don't compile without their required pr's.

**Pull request closed without merging**